source/dagger/gameplay/ping_pong/player_scores.cpp

```diff
@@ -49,7 +49,8 @@ void PlayerScoresSystem::Run()
 
     if (ballOnField == 0)
     {
-        // TODO: set speed to be random in both directions
-        CreatePingPongBall(s_TileSize, ColorRGBA(1, 1, 1, 1), { rand()%10 + 4,rand()%10 + 4,0 },   { 0,rand()%(s_FieldHeight / 2),0 });
+        float xSpeed = rand() % 10 - 4;
```

Comment:
xSpeed may become 0, the ball will freeze if this happens. Try to change your solution